src/transformers/trainer_pt_utils.py

```diff
@@ -1395,3 +1395,211 @@ def get_lr(self):
+            " ensure they are the same across all ranks after initialization"
+        },
+    )
+    cpu_ram_efficient_loading: bool = field(
+        default=False,
+        metadata={
+            "help": "If `True`, only the first process loads the pretrained model checkpoint while all other processes"
+            " have empty weights.  When this setting as `True`, `sync_module_states` also must to be `True`,"
+            " otherwise all the processes except the main process would have random weights leading to unexpected"
+            " behaviour during training."
+        },
+    )
+    activation_checkpointing: bool = field(
+        default=False,
+        metadata={
+            "help": "If `True`, activation checkpointing is a technique to reduce memory usage by clearing activations of"
+            " certain layers and recomputing them during a backward pass. Effectively, this trades extra"
+            " computation time for reduced memory usage."
+        },
+    )
+    state_dict_type: str = field(
+        default="FULL_STATE_DICT",
+        metadata={
+            "help": "State dict type to use. If a string, it must be one of `full_state_dict`, `local_state_dict`, or `sharded_state_dict`"
+        },
+    )
+    xla: bool = field(
+        default=False,
+        metadata={
+            "help": "Whether to use PyTorch/XLA Fully Sharded Data Parallel Training. This is an experimental feature"
+            " and its API may evolve in the future."
+        },
+    )
+    xla_fsdp_settings: Dict = field(
+        default=None,
+        metadata={
+            "help": "The value is a dictionary which stores the XLA FSDP wrapping parameters."
+            " For a complete list of options, please see [here]("
+            " https://github.com/pytorch/xla/blob/master/torch_xla/distributed/fsdp/xla_fully_sharded_data_parallel.py)."
+        },
+    )
+    xla_fsdp_grad_ckpt: bool = field(
+        default=False,
+        metadata={
+            "help": "Will use gradient checkpointing over each nested XLA FSDP wrapped layer. This setting can only be"
+            " used when the xla flag is set to true, and an auto wrapping policy is specified through"
+            " fsdp_min_num_params or fsdp_transformer_layer_cls_to_wrap."
+        },
+    )
+    xla_fsdp_v2: bool = field(
+        default=False,
+        metadata={"help": "Whether to use PyTorch/XLA Fully Sharded Data Parallel Training v2."},
+    )
+
+    @classmethod
+    def from_json_file(cls, json_file):
+        # Check if exists
+        open_file = io.open if os.path.exists(json_file) else open
+        with open_file(json_file, "r", encoding="utf-8") as f:
+            config_dict = json.load(f)
+        # for backward compatibility for keys starting with fsdp_ prefix
+        for k in list(config_dict.keys()):
+            if k.startswith("fsdp_"):
+                v = config_dict.pop(k)
+                config_dict[k[5:]] = v
```

Comment:
### Code change comments


For backward compatibility for keys starting with fsdp_ prefix